dotnet-efcore/EntityFrameworkCoreVsDapper/EntityFrameworkCoreVsDapper/Repositories/PersonsRepositoryDapper.cs

```diff
@@ -11,42 +11,15 @@ public class PersonsRepositoryDapper : IPersonsRepository
         private string ConnectionString = "";
 
         [Benchmark]
-        public List<Person> Get_1000_Persons()
+        [Arguments(1000)]
+        [Arguments(10000)]
+        [Arguments(100000)]
+        public List<Person> Get_X_Persons(int personsNumber)
         {
-            using (var dbConnection = new SqlConnection(ConnectionString))
-            {
-                dbConnection.Open();
-                var items = (List<Person>)dbConnection.Query<Person>("SELECT TOP(1000) * FROM Persons");
-                dbConnection.Close();
+            using var dbConnection = new SqlConnection(ConnectionString);
 
-                return items;
-            }
-        }
-
-        [Benchmark]
-        public List<Person> Get_10000_Persons()
-        {
-            using (var dbConnection = new SqlConnection(ConnectionString))
-            {
-                dbConnection.Open();
-                var items = (List<Person>)dbConnection.Query<Person>("SELECT TOP(10000) * FROM Persons");
-                dbConnection.Close();
-
-                return items;
-            }
-        }
-
-        [Benchmark]
-        public List<Person> Get_100000_Persons()
-        {
-            using (var dbConnection = new SqlConnection(ConnectionString))
-            {
-                dbConnection.Open();
-                var items = (List<Person>)dbConnection.Query<Person>("SELECT TOP(100000) * FROM Persons");
-                dbConnection.Close();
-
-                return items;
-            }
+            dbConnection.Open();
+            return (List<Person>)dbConnection.Query<Person>($"SELECT TOP({personsNumber}) * FROM Persons");
```

Comment:
I may not be correct on this, so please disregard if I am not. I have not used Dapper all that much, but I was thinking that the Query<T>() method returns an IEnumerable<T>. If that is the case, then you should be calling ToList() rather than casting. If you look at the Dapper source code, within the SqlMapper, ther result of the Query call calls to an internal QueryImpl method which involves yield return statements. Maybe there is magic in the compiler that fixes this up, which is why I am mentioning it, but you, the editor and head code reviewer can decide on correctness.